src/uu/mv/src/mv.rs

```diff
@@ -751,3 +845,370 @@ fn is_empty_dir(path: &Path) -> bool {
         Err(_e) => false,
     }
 }
+
+/// Moves a directory from one location to another with progress tracking.
+/// This function assumes that `from` is a directory and `to` does not exist.
+
+/// Returns:
+/// - `Result<u64>`: The total number of bytes moved if successful.
+fn move_dir(
+    from: &Path,
+    to: &Path,
+    progress_bar: Option<&ProgressBar>,
+    vebose_context: Option<VerboseContext<'_>>,
+) -> MvResult<u64> {
+    // The return value that represents the number of bytes copied.
+    let mut result: u64 = 0;
+    let mut error_occurred = false;
+    let mut moved_entries: Vec<(PathBuf, FileType)> = vec![];
+    for dir_entry_result in WalkDir::new(from) {
+        match dir_entry_result {
+            Ok(dir_entry) => {
+                let file_type = dir_entry.file_type();
+                let dir_entry_path = dir_entry.into_path();
+                //comment why this is okay
+                let tmp_to = dir_entry_path.strip_prefix(&from).unwrap();
+                let dir_entry_to = to.join(tmp_to);
+                if file_type.is_dir() {
+                    // check this create all align with gnu
+                    let res = create_all(&dir_entry_to, false);
```

Comment:
Metadata like `rwx` flags aren't copied. It's okay if you decide that this has to be done in a different PR, but this seems like an easy fix, maybe?